index.html

```diff
@@ -1,107 +1,120 @@
+<hr />
+</div>
+<ol class="no-print">
+<li>
+<p>The wallet accounts are generated from a 32 byte seed input as a 64 character long hexadecimal string.</p>
+<p>Click "Use Random" to select a random value using <a href="https://github.com/dchest/tweetnacl-js#naclrandombyteslength">nacl.js</a>.
+<p><label>Seed: <input autocomplete="off" class="param" id="seed" /></label><button id="randomSeed">Use Random</button></p>
+<p><strong>Back up the wallet seed somewhere safe.</strong> You may also print this page after generation.</p>
+<p>Anyone who knows it is able to generate your private keys, thus having access to your funds.</p>
+</li>
+<li>
+<p>Up to 2<sup>32</sup> accounts may be addressed from a single seed. Please select which account to generate a private key/address pair.</p>
+<p>By default, <a href="https://raiwallet.com/">RaiWallet</a> uses the 0th account. The desktop wallet seems to start with 0-2.</a></p>
+<p><label>Account Number: <input class="param" id="accountIndex" type="number" min="0" max="4294967295" value="0" /></p>
+</li>
+<li>
+<p>Click the button to generate a account key/address pair that you may import into other wallet applications.</p>
+<p><button id="generatePair">Generate account pair</button></p>
+</li>
+</ol>
+<hr />
+<ol id="pairs"></ol>
+<script>
+function newPair(seed, accountIndex) {
+  var index = hex_uint8(dec2hex(accountIndex, 4));
+
+  var context = blake2bInit(32);
+  blake2bUpdate(context, seed);
+  blake2bUpdate(context, index);
+
+  var newKey = blake2bFinal(context);
+
+  return {
+    secret: uint8_hex(newKey),
+    address: accountFromHexKey(uint8_hex(nacl.sign.keyPair.fromSecretKey(newKey).publicKey))
+  };
+}
+
+var pairList = document.getElementById('pairs');
+var seedInput = document.getElementById('seed');
+var accountIndexInput = document.getElementById('accountIndex');
+
+document.getElementById('randomSeed').addEventListener('click', function() {
+  seedInput.value = uint8_hex(nacl.randomBytes(32));
+}, false);
+
+document.getElementById('generatePair').addEventListener('click', function() {
+  if(!/[0-9A-F]{64}/i.test(seedInput.value)) {
+    alert('Seed must be 64 character hexadecimal string!');
+    return;
+  }
+
+  var accountIndex = parseInt(accountIndexInput.value, 10);
+  if(accountIndex < 0 || accountIndex >= Math.pow(2,32)) {
+    alert('Account index must be 32 bit unsigned integer!');
+    return;
+  }
+
+  var pair = newPair(hex_uint8(seedInput.value), accountIndex);
+//QRcode:    
+var typeNumber = 4;
+var errorCorrectionLevel = 'L';
+var qrseed = qrcode(typeNumber, errorCorrectionLevel);
+var qrkey = qrcode(typeNumber, errorCorrectionLevel);
+var qraddress = qrcode(typeNumber, errorCorrectionLevel);
+qrseed.addData(seedInput.value);
+qrkey.addData(pair.secret);
+qraddress.addData(pair.address);
+qrseed.make();
+qrkey.make();
+qraddress.make();

```

Comment:
Instead of all this, put a `qrcode` property on the object list below and then in the map function so you can re-use the same code for each field:

```js
.map(function(item) {
    return '<div><dt>' + item.label + ':</dt></div>' +
      '<div><dd>' +
      ('qrcode' in item ? qrcode(4, 'L').addData(item.qrcode).make().createImgTag() : '') +
      '<span class="dont-break-out">' + item.value + '</span>' +
      '</dd></div>';
  })
```

Put some CSS to make the image display decently. Maybe even special coloring around the seed/private key qrcodes than the account address qrcode so that people don't get them mixed up.

Also maybe a checkbox on the top of the page 'Generate QR Codes' or 'Generate QR Codes for Private Keys' that is by-default unchecked. Some people may not want their seed/private key easily scannable by somebody standing next to them.